tests/integration/cocoon/test_cocoon_portfolio_groups.py

```diff
@@ -0,0 +1,324 @@
+import unittest
+from pathlib import Path
+import pandas as pd
+import json
+import lusid
+import lusid.models as models
+from lusidtools import cocoon as cocoon
+from lusidtools.cocoon.utilities import create_scope_id
+from lusidtools import logger
+
+
+class CocoonTestPortfolioGroup(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+
+        cls.scope = create_scope_id()
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+        cls.logger = logger.LusidLogger("debug")
+
+        def create_portfolio_model(code):
+            model = models.CreateTransactionPortfolioRequest(
+                display_name=code,
+                code=code,
+                base_currency="GBP",
+                description="Paper transaction portfolio",
+                created="2020-02-25T00:00:00Z",
+            )
+            return model
+
+        try:
+
+            for code in ["TEST_COM1", "TEST_COM2", "TEST_COM3", "TEST_COM4"]:
+                cls.api_factory.build(
+                    lusid.api.TransactionPortfoliosApi
+                ).create_portfolio(
+                    scope="test_scope_20200225",
+                    transaction_portfolio=create_portfolio_model(code),
+                )
+        except lusid.exceptions.ApiException as e:
+            if e.status == 404:
+                print(f"The portfolio {code} already exists")
+
+    def test_multiple_port_groups_multiple_portfolios(self) -> None:
+
+        """
+        Description:
+        ------------
+        Here we test adding multiple new portfolio groups with multiple portfolios.
+        """
+
+        data_frame = pd.read_csv(
+            Path(__file__).parent.joinpath("data/port_group_tests/portfolio_group.csv")
+        )
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=self.scope,
+            data_frame=data_frame,
+            mapping_required={
+                "code": "PortGroupCode",
+                "display_name": "PortGroupDisplayName",
+            },
+            mapping_optional={"values.scope": "Scope", "values.code": "FundCode",},
+            file_type="portfolio_group",
+            property_columns=[],
+            properties_scope=None,
+        )
+
+        if responses.get("portfolio_groups").get("errors") != []:
```

Comment:
For the first line, I needed to use .get to check if the key exists. The interpreter throws a KeyError if you try and index on a key which does not exists. 
For the second line, we know the key exists (since the code passed the "if" statement) and therefore we can index as normal using [].

Reply:
.get will still throw a key error unless you specify a default, you would need e.g. responses.get("portfolio_groups", {}).get("errors", []) to get what you desire here.